orttraining/orttraining/python/ort_trainer.py

```diff
@@ -1181,14 +1181,18 @@ def eval_step(self, *args, **kwargs):
     def _verify_fully_optimized_model(self, model):
         assert len(model.graph.output) > 0
         # model's first output must be the loss tensor
-        if (
-            model.graph.output[0].type.tensor_type.elem_type != onnx.TensorProto().FLOAT
-            and model.graph.output[0].type.tensor_type.elem_type != onnx.TensorProto().FLOAT16
-            and model.graph.output[0].type.tensor_type.elem_type != onnx.TensorProto().DOUBLE
-            and model.graph.output[0].type.tensor_type.elem_type != onnx.TensorProto().COMPLEX64
-            and model.graph.output[0].type.tensor_type.elem_type != onnx.TensorProto().COMPLEX128
-            and model.graph.output[0].type.tensor_type.elem_type != onnx.TensorProto().BFLOAT16
-        ):
+        if model.graph.output[0].type.tensor_type.elem_type not in {
+            onnx.TensorProto().FLOAT,
+            onnx.TensorProto().FLOAT16,
+            onnx.TensorProto().DOUBLE,
+            onnx.TensorProto().COMPLEX64,
+            onnx.TensorProto().COMPLEX128,
+            onnx.TensorProto().BFLOAT16,
+            onnx.TensorProto().FLOAT8E4M3FN,
+            onnx.TensorProto().FLOAT8E4M3FNUZ,
+            onnx.TensorProto().FLOAT8E5M2,
+            onnx.TensorProto().FLOAT8E5M2FNUZ,
```

Comment:
not necessary to create a TensorProto
```suggestion
            onnx.TensorProto.FLOAT,
            onnx.TensorProto.FLOAT16,
            onnx.TensorProto.DOUBLE,
            onnx.TensorProto.COMPLEX64,
            onnx.TensorProto.COMPLEX128,
            onnx.TensorProto.BFLOAT16,
            onnx.TensorProto.FLOAT8E4M3FN,
            onnx.TensorProto.FLOAT8E4M3FNUZ,
            onnx.TensorProto.FLOAT8E5M2,
            onnx.TensorProto.FLOAT8E5M2FNUZ,
```